### PR TITLE
TSK-1299: Added embedded LDAP to Wildfly example.

### PR DESCRIPTION
**Pull request by holgerhagen**

### CODE REVIEW

**Comment by holgerhagen** [comment]
New Documentation:

- https://taskana.atlassian.net/wiki/spaces/TAS/pages/808091649/Testing+of+the+JBoss+Wildfly+deployment
- https://taskana.atlassian.net/wiki/spaces/TAS/pages/880902174/Wildfly+Application+Server+configuration

**Comment by holgerhagen** [comment]
Sonar:
https://sonarcloud.io/dashboard?branch=tsk1299&id=holgerhagen_taskana

File: rest/taskana-rest-spring-example-common/src/test/java/pro/taskana/rest/security/SpringBootWebSecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+  @Bean
+  public FilterRegistrationBean<CorsFilter> corsFilter() {
+    final UrlBasedCorsConfigurationSource source = new UrlBasedCorsConfigurationSource();
+    CorsConfiguration config = new CorsConfiguration();
+    config.setAllowCredentials(true);
+    config.addAllowedOrigin("*");
+    config.addAllowedHeader("*");
+    config.addAllowedMethod("*");
+    config.addAllowedMethod("POST");
```
[the remark this answers is not in the document]
Comment (holgerhagen) [reply rest/taskana-rest-spring-example-common/src/test/java/pro/taskana/rest/security/SpringBootWebSecurityConfig.java]: done

File: rest/taskana-rest-spring-example-wildfly/src/main/java/pro/taskana/wildfly/security/WildflyWebSecurityConfig.java [hunk 2]
```diff
@@ -1,141 +1,65 @@
...
-      }
-    };
+  public FilterRegistrationBean<CorsFilter> corsFilter() {
+    final UrlBasedCorsConfigurationSource source = new UrlBasedCorsConfigurationSource();
+    CorsConfiguration config = new CorsConfiguration();
+    config.setAllowCredentials(true);
+    config.addAllowedOrigin("*");
+    config.addAllowedHeader("*");
+    config.addAllowedMethod("*");
+    config.addAllowedMethod("POST");
```
[the remark this answers is not in the document]
Comment (holgerhagen) [reply rest/taskana-rest-spring-example-wildfly/src/main/java/pro/taskana/wildfly/security/WildflyWebSecurityConfig.java]: done

File: rest/taskana-rest-spring-example-boot/src/main/java/pro/taskana/rest/security/SpringBootWebSecurityConfig.java [hunk 3]
```diff
@@ -0,0 +1,180 @@
+package pro.taskana.rest.security;
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring-example-boot/src/main/java/pro/taskana/rest/security/SpringBootWebSecurityConfig.java:1]: I've never done something with websecurityconfig. 
Is something special done here?

File: rest/taskana-rest-spring-example-wildfly/pom.xml [hunk 4]
```diff
@@ -95,15 +81,25 @@
...
             <scope>test</scope>
         </dependency>
-        <!-- Brought in via WildFly Swarm bom -->
         <dependency>
-            <groupId>io.thorntail</groupId>
-            <artifactId>arquillian</artifactId>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <scope>test</scope>
+        </dependency>
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring-example-wildfly/pom.xml:90]: why did you add junit4 as a dependency? We agreed on using junit5

File: rest/taskana-rest-spring-example-wildfly/pom.xml [hunk 5]
```diff
@@ -95,15 +81,25 @@
...
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-test</artifactId>
             <scope>test</scope>
         </dependency>
+        <!-- Brought in via WildFly Swarm bom -->
+        <dependency>
+            <groupId>org.wildfly.arquillian</groupId>
+            <artifactId>wildfly-arquillian-container-managed</artifactId>
+            <version>2.2.0.Final</version>
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring-example-wildfly/pom.xml:100]: remove version (inherited from bom)
